File: easy/SortArrayByParity.js
// Given an integer array nums, move all the even integers at the beginning of the array followed by all the odd integers.

// Return any array that satisfies this condition.

const sortArrayByParity = function(nums) {
  return nums.sort((a,b) => {
    if (a % 2 === 0 && b % 2 !== 0) {
      return -1;
    } else if (a % 2 !== 0 && b % 2 === 0) {
      return 1;
    } else {
      return 0;
    }
  })
};

console.log(sortArrayByParity([3,1,2,4]))

File: easy/SortArrayByParity.py
# Given an integer array nums, move all the even integers at the beginning of the array followed by all the odd integers.

# Return any array that satisfies this condition.

def sortArrayByParity(nums):
    i = 0
    j = len(nums) - 1
    while i < j:
        if nums[i] % 2 != 0:
            while i < j:
                if nums[j] % 2 == 0:
                    nums[i], nums[j] = nums[j], nums[i]
                    break
                j -= 1
        i += 1
    return nums


print(sortArrayByParity([3, 1, 2, 4]) == [4, 2, 1, 3])
